src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/services/WarningTaskHandlerTest.java

```diff
@@ -1,30 +1,127 @@
 package uk.gov.hmcts.reform.wacaseeventhandler.services;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.wacaseeventhandler.clients.WorkflowApiClientToWarnTask;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.cancellationtask.CancellationCorrelationKeys;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.cancellationtask.CancellationEvaluateRequest;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.DmnStringValue;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.EvaluateDmnRequest;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.EvaluateDmnResponse;
 import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.EventInformation;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.ProcessVariables;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.SendMessageRequest;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.warningtask.WarningEvaluateResponse;
 import uk.gov.hmcts.reform.wacaseeventhandler.handlers.WarningTaskHandler;
 
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
+@ExtendWith(MockitoExtension.class)
 class WarningTaskHandlerTest {
 
-    private final WarningTaskHandler handlerService = new WarningTaskHandler();
+    private static final String DMN_NAME = "wa-task-cancellation-ia-asylum";
+    public static final String WARN_TASKS_MESSAGE_NAME = "warnProcess";
+
+
+    @Mock
+    private WorkflowApiClientToWarnTask workflowApiClientToWarnTask;
+
+    @Captor
+    private ArgumentCaptor<SendMessageRequest<ProcessVariables, CancellationCorrelationKeys>> sendMessageRequestCaptor;
+
+    @InjectMocks
+    private WarningTaskHandler handlerService;
+
+    private final EventInformation eventInformation = EventInformation.builder()
+        .eventId("some event id")
+        .newStateId("some post state")
+        .previousStateId("some previous state")
+        .jurisdictionId("ia")
+        .caseTypeId("asylum")
+        .caseReference("some case reference")
+        .dateTime(LocalDateTime.now())
+        .build();
 
     @Test
     void evaluateDmn() {
```

Comment:
I'd recommend having more descriptive names for tests... e.g. `should_call_evaluateDmn_and_return_empty_list_response`

The idea being that someone can just read the test title to know what's going on in it.